### ISSUE TITLE
SC2283 ("spaces around = to assign") doesn't trigger with readonly or declare

### ISSUE THREAD

**Issue by wimax-grapl** [koalaman/shellcheck#2279]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2283
- My shellcheck version (`shellcheck --version` or "online"): online
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash

# This should be a shellcheck error
readonly A = 1
echo "$A"

# Counterexample (it works without readonly)
B = 2
echo "$B"

# UPDATE: You should probably do this check for declare as well!
declare -r C = 3
echo "$C"
```

#### Here's what shellcheck currently says:
```
$ shellcheck myscript
 
Line 8:
B = 2
  ^-- SC2283: Remove spaces around = to assign (or use [ ] to compare, or quote '=' if literal).

```

#### Here's what I wanted or expected to see:
```
Line 4:
readonly A = 1
           ^-- SC2283: Remove spaces around = to assign (or use [ ] to compare, or quote '=' if literal).
Line 8:
B = 2
  ^-- SC2283: Remove spaces around = to assign (or use [ ] to compare, or quote '=' if literal).

and the same for C
```

**Comment by wittekm** [comment]
Sidenote: this appears to be a regression, a coworker says it detected it in 0.5.0

**Comment by koalaman** [comment]
These warnings were a victim of fbb14d6b384a65ee16781966d9d240db9ed7b644, which improved the robustness and flexibility of warnings about e.g. `${var} = value` by moving the logic from the parser to the analyzer. However, this rewrite did not account for the other places where the assignment parser was used: after `declare`, `export`, `local`, `readonly` and `typeset`.

This fix re-adds a simple check for words starting with `=` or `+=` for these commands. Good catch, thanks!

I investigated whether it could also benefit from other assignment bugs like `export $var=$value`, but all the cases I found were using this correctly.

**Comment by wimax-grapl** [comment]
Thank you Vidar!!!!!
